src/runtimes/node/utils/zip.ts

```diff
@@ -116,7 +120,7 @@ const createZipArchive = async function ({
   const userNamespace = hasEntryFileConflict ? DEFAULT_USER_SUBDIRECTORY : ''
 
   if (needsEntryFile) {
```

Comment:
There's a caveat with the current PR in that we won't generate the layer code if `needsEntryFile` is `false`. We'll need to figure that out separately.